docs/Developers/NextGen_Core.rst

```diff
@@ -0,0 +1,286 @@
+NextGen-Core
+============
+
+This document provides a brief overview of coala's NextGen-Core.
+coala's NextGen-Core comes with the promise of resolving the dependencies
```

Comment:
maybe instead of mentioning this specific "dependency" topic, we could do something like
> ... comes with the promise to lift many limitations of the existing core and better efficiency and performance.

Up to you :)